[webkitscmpy] Generalize pull-request title generation.

JonWBedard opened the pull request:
#### 5473ad2add269a8914f5bf8b9f3e3a7375c01d9e
<pre>
[webkitscmpy] Generalize pull-request title generation.
<a href="https://bugs.webkit.org/show_bug.cgi?id=232463">https://bugs.webkit.org/show_bug.cgi?id=232463</a>
&lt;rdar://problem/84784354 &gt;

Reviewed by Dewei Zhu.

* Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.title_for): Compute pull-request title for a set of commits.
(PullRequest.main): Generalize pull-request title generation.
</pre>

rniwa commented:
Closing.

dewei-zhu commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
```diff
@@ -81,6 +81,18 @@ def create_commit(cls, args, repository, **kwargs):
+        title = commits[0].message.splitlines()[0]
+        for commit in commits[1:]:
+            title_candidate = commit.message.splitlines()[0]
+            while title and not title_candidate.startswith(title):
```
This is O(n^2) way to calculate common-prefix [note str.startswith is O(n)], where can we directly implement it with O(n).

JonWBedard replied:
Turns out, `os` implements this anyways, so we can just use that.

JonWBedard committed:
[webkitscmpy] Generalize pull-request title generation.
https://bugs.webkit.org/show_bug.cgi?id=232463
<rdar://problem/84784354>

Reviewed by Dewei Zhu.

* Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.title_for): Compute pull-request title for a set of commits.
(PullRequest.main): Generalize pull-request title generation.
